pkg/netpol/eval/internal/k8s/adminnetpol.go

```diff
@@ -352,120 +595,51 @@ func determineConnResByAction(action string, isBANPrule bool) (res ANPRulesResul
-	if p.PeerType() == IPBlockType {
-		// adminNetworkPolicy is a cluster level resource which selects peers with their namespaceSelectors and podSelectors only,
-		// so it might not select IPs
-		return false, nil
-	}
-	if !anp.adminPolicyAffectsDirection(isIngress) {
-		return false, nil
-	}
-	// check if the subject selects the given peer
-	errTitle := fmt.Sprintf("%s %q: ", anpErrTitle, anp.Name)
-	return subjectSelectsPeer(anp.Spec.Subject, p, errTitle)
-}
-
-// adminPolicyAffectsDirection returns whether the anp affects the given direction or not.
-// anp affects a direction, if its spec has rules on that direction
-func (anp *AdminNetworkPolicy) adminPolicyAffectsDirection(isIngress bool) bool {
-	if isIngress {
-		// ANPs with no ingress rules do not affect ingress traffic.
-		return len(anp.Spec.Ingress) > 0
-	}
-	// ANPs with no egress rules do not affect egress traffic.
-	return len(anp.Spec.Egress) > 0
-}
-
-const anpErrTitle = "admin network policy"
-
-// anpErr returns string format of an error in a rule in admin netpol
-func (anp *AdminNetworkPolicy) anpRuleErr(ruleName, description string) error {
-	return fmt.Errorf("%s %q: %s %q: %s", anpErrTitle, anp.Name, ruleErrTitle, ruleName, description)
-}
-
-// GetIngressPolicyConns returns the connections from the ingress rules selecting the src in spec of the adminNetworkPolicy
-func (anp *AdminNetworkPolicy) GetIngressPolicyConns(src, dst Peer) (*PolicyConnections, error) {
-	res := NewPolicyConnections()
-	for _, rule := range anp.Spec.Ingress { // rule is apisv1a.AdminNetworkPolicyIngressRule
-		rulePeers := rule.From
-		rulePorts := rule.Ports
-		if err := updateConnsIfIngressRuleSelectsPeer(rulePeers, rulePorts, src, dst, res, string(rule.Action), false); err != nil {
-			return nil, anp.anpRuleErr(rule.Name, err.Error())
-		}
+// isIPv6Cidr returns if the cidr is in IPv6 format
+func isIPv6Cidr(cidr apisv1a.CIDR) (bool, error) {
+	_, ipNet, err := net.ParseCIDR(string(cidr))
+	if err != nil {
+		return false, err
 	}
-	return res, nil
+	ip := ipNet.IP
+	// if the IP is IPv6, return true
+	return ip.To4() == nil, nil
 }
 
-// GetEgressPolicyConns returns the connections from the egress rules selecting the dst in spec of the adminNetworkPolicy
-func (anp *AdminNetworkPolicy) GetEgressPolicyConns(dst Peer) (*PolicyConnections, error) {
-	res := NewPolicyConnections()
-	for _, rule := range anp.Spec.Egress { // rule is apisv1a.AdminNetworkPolicyEgressRule
-		rulePeers := rule.To
-		rulePorts := rule.Ports
-		if err := updateConnsIfEgressRuleSelectsPeer(rulePeers, rulePorts, dst, res, string(rule.Action), false); err != nil {
-			return nil, anp.anpRuleErr(rule.Name, err.Error())
+// rulePeersReferencedNetworks returns a list of IPBlocks representing the CIDRs referenced by the given rulePeers' Networks field.
+func rulePeersReferencedNetworks(rulePeers []apisv1a.AdminNetworkPolicyEgressPeer) ([]*netset.IPBlock, error) {
```

Comment:
rename to `rulePeersReferencedIPBlocks`, as the function returns IPBlock objects.

Reply:
done